### PR TITLE
Abstract and Condense API Accesses

### PR DESCRIPTION
**Pull request by amit-sawhney**
## Status:

:rocket: Ready

## Summary

Creates hooks and consolidates global states to access the data for a user's address. 

## Changes

- Create `useBallotData()` hook

### CODE REVIEW

File: client/src/components/ui/Autocomplete.tsx [hunk 1]
```diff
@@ -99,25 +168,27 @@ export const SearchAutocomplete: FC<SearchAutoCompleteProps> = ({
           <MicIcon />
         </InputRightElement>
       </InputGroup>
+      {error && (
```
Comment (ayanmallik) [inline client/src/components/ui/Autocomplete.tsx:171]: Great catch - this should hopefully resolve any issues with invalid addresses that don't have proper ocd id's:

image

File: client/src/contexts/BallotContext.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      {children}
+    </BallotDataContext.Provider>
+  );
+};
+
+/**
+ *
+ * @returns
+ */
+export const useBallotData = (): BallotDataContextInterface => {
```
Comment (ayanmallik) [inline client/src/contexts/BallotContext.tsx:115]: Thanks for putting all the ballot data together and streamlining it through this context provider - we'll work with Arman now to integrate it into his ballot page.

**Review by ayanmallik — APPROVED** [review]
Awesome job, Amit! Once again, I can't imagine what this repo would look like without your clutch infrastructure. We'll work on integrating/debugging the `useBallotData()` hook right away!

**Comment by code-arman** [comment]
change the pr status if its ready :)

**Pull request merged**